File: leetcode/2279-maximum-bags-with-full-capacity-of-rocks/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 210 ms (81.98%)
     * MEMORY: 84.7 MB (79.56%)
    */
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int additionalRocks) {
        for (int i = 0; i < capacity.size(); i++) {
            capacity[i] -= rocks[i];
        }
        sort(capacity.begin(), capacity.end());
        int cnt = 0;
        int result = count(capacity.begin(), capacity.end(), 0);
        for (int i = result; i < capacity.size(); i++) {
            cnt += capacity[i];
            if (cnt <= additionalRocks) { result += 1; }
            else { break; }
        }
        return result;
    }
};

File: leetcode/2279-maximum-bags-with-full-capacity-of-rocks/solution.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        """
        RUNTIME: 929 ms (96.5%)
        MEMORY: 22.1 MB (87.83%)
        """
        for i in range(len(capacity)):
            capacity[i] -= rocks[i]
        capacity.sort(reverse=False)
        cnt = 0
        result = capacity.count(0)
        capacity[:] = capacity[result:]
        for i in capacity:
            cnt += i
            if cnt <= additionalRocks:
                result += 1
            else:
                break
        return result
